lib/HTFeed/Stage/Pack.pm

```diff
@@ -73,24 +76,43 @@ sub run{
 
 sub zip{
     my $self = shift;
-    my ($zip_stage,$other_options,$zip_path,$pt_objid) = @_;
+    my ($zip_stage, $other_options, $zip_path, $pt_objid) = @_;
 
+    my $start_size = $self->dir_size($zip_stage);
     get_logger()->trace("Packing $zip_stage/$pt_objid to $zip_path");
     my $zipret = system("cd '$zip_stage'; zip -q -r $other_options '$zip_path' '$pt_objid'");
 
-    if($zipret) {
-        $self->set_error('OperationFailed',operation=>'zip',detail=>'Creating zip file',exitstatus=>$zipret,file=>$zip_path);
+    if ($zipret) {
+        $self->set_error(
+	    'OperationFailed',
+	    operation  => 'zip',
+	    detail     => 'Creating zip file',
+	    exitstatus => $zipret,
+	    file       => $zip_path
+	);
         return;
     } else {
-
         $zipret = system("unzip -qq -t '$zip_path'");
-
-        if($zipret) {
-            $self->set_error('OperationFailed',operation=>'unzip',exitstatus=>$zipret,file=>$zip_path,detail=>'Verifying zip file');
+        if ($zipret) {
+            $self->set_error(
+		'OperationFailed',
+		operation  => 'unzip',
+		detail     => 'Verifying zip file',
+		exitstatus => $zipret,
+		file       => $zip_path
+	    );
             return;
-        }        
+        }
     }
 
+    # Get size after zipping and report size delta as a job metric for Pack.
+    my $end_size = -s $zip_path;
+    my $delta_size = $start_size - $end_size;
+    $self->{jm}->add("Pack_bytes", $delta_size);
```

Comment:
Might rather record the number of (uncompressed) bytes read and the (compressed) bytes written? The delta size is likely to generally be rather small and I'm not sure terribly meaningful

Reply:
Made an issue of this. Resolving convo.